server/main_server/src/main.cpp

```diff
@@ -1491,6 +1518,62 @@ Environment Variables:
                 const auto ec = rcModDataObjMeta(static_cast<RcComm*>(conn), &input);
                 log_server::debug("{}: rcModDataObjMeta returned [{}]", __func__, ec);
             }
+#else
+            irods::experimental::client_connection conn;
+            irods::access_time_manager::access_time_data data;
+            nlohmann::json::array_t updates;
+
+            const auto update_count = irods::access_time_manager::number_of_queued_updates();
+            for (std::size_t i = 0; i < update_count; ++i) {
+                if (!irods::access_time_manager::try_dequeue(data)) {
+                    break;
+                }
+
+                updates.push_back(nlohmann::json{
+                    {"data_id", data.data_id},
+                    {"replica_number", data.replica_number},
+                    {"atime", data.last_accessed}
+                });
+            }
+
+            if (updates.empty()) {
+                return;
+            }
+
+            // Group elements by data id and replica number in descending order.
+            std::sort(std::begin(updates), std::end(updates), [](const auto& _lhs, const auto& _rhs) {
+                const auto& lhs_data_id = _lhs.at("data_id");
+                const auto& rhs_data_id = _rhs.at("data_id");
+                if (lhs_data_id > rhs_data_id) {
+                    return true;
+                }
+
+                if (lhs_data_id == rhs_data_id) {
+                    const auto& lhs_replica_number = _lhs.at("replica_number");
+                    const auto& rhs_replica_number = _rhs.at("replica_number");
+                    if (lhs_replica_number > rhs_replica_number) {
+                        return true;
+                    }
+
+                    if (lhs_replica_number == rhs_replica_number) {
+                        return _lhs.at("atime") > _rhs.at("atime");
+                    }
+                }
+
+                return false;
+            });
+
+            // Reduce updates such that each replica is updated once. On completion, the JSON
```

Comment:
is it also possible to replace any data_id:replica_number tuple in the queue upon insertion to the queue?   so it's really just a big map?  can discuss.

Reply:
Not without implementing our own map. Given this is all internal and UGM is approaching, I think the current implementation is acceptable. It will at least give us a solid base to start from.

We could/should consider capturing some statistics for the atime updates - e.g. how often is is full, enqueue/dequeue frequency, etc. Of course, that implies we have somewhere to store that information.

Reply:
agreed on metrics... could just dump a little stats file alongside the server... perhaps we grow a little 'write_down_stats' flag in the config...

or we wait for our metrics framework to appear...